src/joystick/windows/SDL_rawinputjoystick.c

```diff
@@ -714,6 +713,12 @@ RAWINPUT_AddDevice(HANDLE hDevice)
     device->is_xinput = SDL_TRUE;
     device->is_xboxone = GuessControllerType(device->vendor_id, device->product_id) == k_eControllerType_XBoxOneController;
 
+    /* Get HID Top-Level Collection Preparsed Data */
+    size = 0;
+    CHECK(GetRawInputDeviceInfoA(hDevice, RIDI_PREPARSEDDATA, NULL, &size) != (UINT)-1);
+    CHECK(device->preparsed_data = (PHIDP_PREPARSED_DATA)SDL_calloc(size, sizeof(BYTE)));
+    CHECK(GetRawInputDeviceInfoA(hDevice, RIDI_PREPARSEDDATA, device->preparsed_data, &size) != (UINT)-1);
```

Comment:
What is the advantage of this over `HidD_GetPreparsedData()`?

Reply:
RawInput API provides pp data and I think it is clearer to request it like that. Just for consistency.